### PR TITLE
typora: fix

### PR DESCRIPTION
**Pull request by liberodark**
Fix partially : https://github.com/NixOS/nixpkgs/issues/362022
- Fix open .md files
- Fix LibGL issue
- Add Launcher (Inspired by AUR)

Launcher add possibility to set : 

```
# ~/.config/typora-flags.conf
--disable-gpu
--no-sandbox
```

Only is missing spell support : 
https://support.typora.io/Spellcheck/
https://github.com/typora/dictionaries

If you have idea for fix this ?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by liberodark** [comment]
Also if you can backport this PR.

**Comment by npulidomateo** [comment]
@liberodark Thanks a lot for looking into this.

The spellcheck issue seems to be a permissions problem. Typora creates the folder `~/.config/Typora/typora-dictionaries` and then tries to download the dictionaries there. 

In my system that folder is created without write permissions (only read and execute permissions for everybody). If you give it write permissions then Typora installs the dictionaries with no problem.

File: pkgs/by-name/ty/typora/package.nix [hunk 1]
```diff
@@ -73,7 +75,22 @@ let
...
+    name = "typora-launcher";
+    executable = true;
+    text = ''
+      #!${stdenv.shell}
+      XDG_CONFIG_HOME=''${XDG_CONFIG_HOME:-~/.config}
+
+      if [ -f "$XDG_CONFIG_HOME/typora-flags.conf" ]; then
+        TYPORA_USER_FLAGS="$(sed 's/#.*//' "$XDG_CONFIG_HOME/typora-flags.conf" | tr '\n' ' ')"
+      fi
+
```
Comment (npulidomateo) [inline pkgs/by-name/ty/typora/package.nix:92]: We could add here:

```
mkdir -p $XDG_CONFIG_HOME/Typora/typora-dictionaries
chmod u+w $XDG_CONFIG_HOME/Typora/typora-dictionaries
```

**Comment by liberodark** [comment]
@npulidomateo Thank you 

What do you think with this : 

```
launchScript = writeTextFile {
  name = "typora-launcher";
  executable = true;
  text = ''
    #!${stdenv.shell}
    
    # Configuration directory setup
    XDG_CONFIG_HOME=''${XDG_CONFIG_HOME:-~/.config}
    TYPORA_CONFIG_DIR="$XDG_CONFIG_HOME/Typora"
    TYPORA_DICT_DIR="$TYPORA_CONFIG_DIR/typora-dictionaries"
    
    # Remove existing read-only dictionary directory if it exists
    if [ -d "$TYPORA_DICT_DIR" ]; then
      rm -rf "$TYPORA_DICT_DIR"
    fi
    
    # Create config directories with proper permissions
    mkdir -p "$TYPORA_CONFIG_DIR"
    mkdir -p "$TYPORA_DICT_DIR"
    chmod 755 "$TYPORA_CONFIG_DIR"
    chmod 777 "$TYPORA_DICT_DIR"
    
    # Read user flags if they exist
    if [ -f "$XDG_CONFIG_HOME/typora-flags.conf" ]; then
      TYPORA_USER_FLAGS="$(sed 's/#.*//' "$XDG_CONFIG_HOME/typora-flags.conf" | tr '\n' ' ')"
    fi
    
    exec ${typoraFHS}/bin/typora-fhs "$@" $TYPORA_USER_FLAGS
  '';
};
```

**Comment by npulidomateo** [comment]
@liberodark  Sorry just read this. I like your suggestion better than mine. Let's go with yours.

**Comment by npulidomateo** [comment]
@liberodark I checked again and Typora seems to create `TYPORA_CONFIG_DIR` already with write permission, so we only need to address `TYPORA_DICT_DIR`. I just tested it and `chmod 755 "$TYPORA_DICT_DIR"` already seems to work so `777` might be overkill?

Thanks again and good night!

**Review by npulidomateo — APPROVED** [review]
Looks good to me.

**Review by chewblacka — APPROVED** [review]
LGTM

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**